### ISSUE TITLE
proposal for tooling

### ISSUE THREAD

**Issue by kkasravi** [kubeflow/kubeflow#250]
i wanted to submit a proposal for tooling under community but wasn't sure of the process or where the template is. @jlewi i can send you a markdown document if you would like (and close this issue)

**Comment by jlewi** [comment]
Can you be more specific? What sort of tooling is this for?

**Comment by kkasravi** [comment]
Create a tool similar to ks but kubeflow centric that automates much of the route sequence of ks commands as well as enabling easy deployment in the target pod or allowing the data scientist to debug locally (eg telepresence, pycharm, goland/dlv, etc).
```
Usage: /usr/local/bin/kf [OPTIONS] COMMANDS
 OPTIONS:
   -h       | --help
   -v       | --version
   -V       | --verbose
 COMMANDS:
   create   [-h] namespace
   batch    [-h] name
   debug    [-h] [-b] [-bn] [-t] [component]
   deploy   [-h] [-d] [-t] [component]
   logs     [-h] [component]
   remove   [-h] [component]
   shell    [-h] [component]
   snapshot [-h] [-t tag] [component]
   status   [-h] [component]
   train    [-h] name
   stream   [-h] name
   whoami   [-h]

 where [component] is one of train|stream|batch
```
 The proposal would need to go into detail of each subcommand and be evaluated against #87

**Comment by aronchick** [comment]
This is a terrific idea! Love it!

**Comment by jlewi** [comment]
Template is here.
https://github.com/kubeflow/community/blob/master/proposals/TEMPLATE.md

@kkasravi Do you want to put a proposal together?

/assign @kkasravi

**Comment by kkasravi** [comment]
@jlewi yes. Is it ok to vet the proposal with you a bit beforehand before opening a PR? Also can i get on the calendar so i can go over the proposal / demo?

**Comment by kkasravi** [comment]
see https://github.com/kubeflow/community/pull/54

**Comment by kkasravi** [comment]
implemented by #2091
